terra_layer/tests/test_style.py

```diff
@@ -0,0 +1,386 @@
+from django.contrib.gis.geos import Point
+from django.test import TestCase
+
+from terra_layer.models import Layer
+
+from django_geosource.models import PostGISSource
+from geostore.models import Feature
+
+from terra_layer import style
+
+import random
+
+
+class StyleTestCase(TestCase):
+    def setUp(self):
+        self.source = PostGISSource.objects.create(
+            name="test",
+            db_name="test",
+            db_password="test",
+            db_host="localhost",
+            geom_type=1,
+            refresh=-1,
+        )
+        self.layer = Layer.objects.create(
+            source=self.source, name="foo", uuid="91c60192-9060-4bf6-b0de-818c5a362d89",
+        )
+
+    def _feature_factory(self, geo_layer, **properties):
+        return Feature.objects.create(
+            layer=geo_layer,
+            geom=Point(-1.560408, 47.218658),
+            properties=properties,
+        )
+
+    def test_min_max(self):
+        geo_layer = self.source.get_layer()
+        self._feature_factory(geo_layer, a=1),
+        self._feature_factory(geo_layer, a=2),
+
+        self.assertEqual(style.min_max(geo_layer, 'a'), [1.0, 2.0])
+
+    def test_empty(self):
+        geo_layer = self.source.get_layer()
+        geo_layer.layer_style_wizard = {}
+        geo_layer.save()
+
+        # Make a random change on the layer before save
+        self.layer.name = 'foobar'
+        self.layer.save()
+        self.assertEqual(self.layer.layer_style, {})
+
+    def test_1feature(self):
+        geo_layer = self.source.get_layer()
+        self._feature_factory(geo_layer, a=1),
+
+        self.layer.layer_style_wizard = {}
+        self.layer.save()
+
+        self.assertEqual(self.layer.layer_style, {})
+
+    graduated_fail = {
```

Comment:
Put this in top of the file and capitalize like constant.

Reply:
Why the "fail" in name ?

Reply:
Removed